app/src/main/java/net/programmierecke/radiodroid2/StationSaveManager.java

```diff
@@ -88,6 +88,16 @@ public DataRadioStation getPreviousById(String id) {
         return listStations.get(listStations.size()-1);
     }
 
+    public void move(int fromPos, int toPos) {
+        if (fromPos < toPos) {
+            for (int i = fromPos; i < toPos; i++)
+                Collections.swap(listStations, i, i + 1);
+        } else {
+            for (int i = fromPos; i > toPos; i--)
+                Collections.swap(listStations, i, i - 1);
+        }
```

Comment:
That's quite an elaborate way to do this =)
It would be better to use this:
```
        if (fromPos <= toPos) {
            Collections.rotate(listStations.subList(fromPos, toPos + 1), -1);
        } else {
            Collections.rotate(listStations.subList(toPos, fromPos + 1), 1);
        }
```

Reply:
Ah, thanks! I wasn't even aware of rotate.